platform/x11/godot_x11.cpp

```diff
@@ -26,13 +26,17 @@
 /* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
 /* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
 /*************************************************************************/
+#include <unistd.h>
 #include "main/main.h"
 #include "os_x11.h"
 
 int main(int argc, char* argv[]) {
 
 	OS_X11 os;
 
+	char *cwd = (char*)malloc(PATH_MAX);
+    getcwd(cwd, PATH_MAX);
```

Comment:
whoopsy, fixed the formatting

As for the String, this is a C function, it requires a char* array of known size. I don't think String provides that interface.